Sources/Action/Action.swift

```diff
@@ -98,19 +97,14 @@ public final class Action<Input, Element> {
             .of(notEnabledError, underlyingError)
             .merge()
 
-        let executionStart = executionObservables
-        let executionEnd = executionObservables
-            .flatMap { observable -> Observable<Void> in
-                return observable
-                    .flatMap { _ in Observable<Void>.empty() }
-                    .concat(Observable.just())
-                    .catchErrorJustReturn()
+        executing = executionObservables.flatMap {
+                Observable.concat([
+                    Observable.just(true),
+                    $0.ignoreElements().map { _ in true }.catchError { _ in Observable.empty() },
+                    Observable.just(false)])
```

Comment:
Putting these operations here looks nice 👍 

I think it might be better to replace `ignoreElements().map { _ in true }` with `flatMap { _ in Observable<Bool>.empty()` if it is possible, because the former returns unrelated value `true` but the latter doesn't.

Additionally, naming `$0.ignoreElements().map { _ in true }.catchError { _ in Observable.empty() }` may improve readability.

```swift
executing = executionObservables
    .flatMap { execution -> Observable<Bool> in
        let execution = execution
            .flatMap { _ in Observable<Bool>.empty() }
            .catchError { _ in Observable.empty() }

        return Observable.concat([
            Observable.just(true),
            execution,
            Observable.just(false)])
    }
    .startWith(false)
    .shareReplay(1)
```